File: Samples/Silverlight_VB/EntitySpaces.Proxies.Silverlight/Proxies/Shippers.vb
'===============================================================================
'                   EntitySpaces Studio by EntitySpaces, LLC
'            Persistence Layer and Business Objects for Microsoft .NET
'            EntitySpaces(TM) is a legal trademark of EntitySpaces, LLC
'                         http://www.entityspaces.net
'===============================================================================
' EntitySpaces Version : 2012.1.0930.0
' EntitySpaces Driver  : SQL
' Date Generated       : 9/23/2012 6:16:24 PM
'===============================================================================

Imports System
Imports System.Collections.Generic
Imports System.Collections.ObjectModel
Imports System.Text
Imports System.Xml.Serialization
Imports System.Runtime.Serialization

Imports EntitySpaces.DynamicQuery

Namespace Proxies

	<DataContract(Namespace:="http://tempuri.org/", Name:="ShippersCollection")> _
	<XmlType(TypeName:="ShippersCollectionProxyStub")> _	
	Partial Public Class ShippersCollectionProxyStub 

		Public Sub New() 
		
		End Sub
		
		<DataMember(Name:="Collection", EmitDefaultValue:=False)> _
		Public Collection As New ObservableCollection (Of ShippersProxyStub)
		
		Public Function IsDirty() As Boolean
			For Each obj As ShippersProxyStub In Collection
				If obj.IsDirty() Then
					Return True
				End If
			Next
			Return False
		End Function
	

	End Class
	
	<DataContract(Namespace:="http://tempuri.org/", Name:="Shippers")> _
	<XmlType(TypeName:="ShippersProxyStub")> _	
	Partial Public Class ShippersProxyStub
		Implements System.ComponentModel.INotifyPropertyChanged
		
		Public Sub New()
			Me.esRowState = "Added"
		End Sub
		
		Public Function IsDirty() As Boolean
			Return If(esRowState <> "Unchanged", True, False)
		End Function
		
		Public Event PropertyChanged As System.ComponentModel.PropertyChangedEventHandler Implements System.ComponentModel.INotifyPropertyChanged.PropertyChanged
		
		Protected Sub RaisePropertyChanged(ByVal propertyName As String)
			Dim propertyChanged As System.ComponentModel.PropertyChangedEventHandler = Me.PropertyChangedEvent
			If (Not (propertyChanged) Is Nothing) Then
				propertyChanged(Me, New System.ComponentModel.PropertyChangedEventArgs(propertyName))
			End If
		End Sub
		
		<DataMember(Name:="a0", Order:=1, EmitDefaultValue:=False)> _
        Public Property ShipperID As Nullable(Of System.Int32)
			Get
				Return _ShipperID
			End Get
			Set(ByVal value As Nullable(Of System.Int32))
				If Not _ShipperID.Equals(value) Then
					_ShipperID = value
					SetDirty("ShipperID")
					RaisePropertyChanged("ShipperID")
				End If
			End Set
		End Property
		Private _ShipperID As Nullable(Of System.Int32)
		<DataMember(Name:="a1", Order:=2, EmitDefaultValue:=False)> _
        Public Property CompanyName As System.String
			Get
				Return _CompanyName
			End Get
			Set(ByVal value As System.String)
				If Not String.Compare(_CompanyName, value) = 0 Then
					_CompanyName = value
					SetDirty("CompanyName")
					RaisePropertyChanged("CompanyName")
				End If
			End Set
		End Property
		Private _CompanyName As System.String
		<DataMember(Name:="a2", Order:=3, EmitDefaultValue:=False)> _
        Public Property Phone As System.String
			Get
				Return _Phone
			End Get
			Set(ByVal value As System.String)
				If Not String.Compare(_Phone, value) = 0 Then
					_Phone = value
					SetDirty("Phone")
					RaisePropertyChanged("Phone")
				End If
			End Set
		End Property
		Private _Phone As System.String
		
		

		<DataMember(Name:="a10000", Order:=10000)> _
		Public Property esRowState As String
			Get
				Return Me._esRowState
			End Get
			Set
				If Not String.Compare(_esRowState, value) = 0 Then
					Me._esRowState = value
					RaisePropertyChanged("esRowState")
				End If
			End Set
		End Property
		Private _esRowState As String = "Unchanged"
		
		<DataMember(Name := "a10001", Order := 10001, EmitDefaultValue := False)> _		
		Public Property ModifiedColumns() As List(Of String)
			Get
				If _ModifiedColumns Is Nothing Then
					_ModifiedColumns = New List(Of String)()
				End If
				Return _ModifiedColumns
			End Get
			Set(ByVal value As List(Of String))
				_ModifiedColumns = New List(Of String)(value)
			End Set
		End Property
		Private _ModifiedColumns As List(Of String)

		<DataMember(Name := "a10002", Order := 10002, EmitDefaultValue := False)> _
		Public Property esExtraColumns() As Dictionary(Of String, Object)
			Get
				If _ExtraColumns Is Nothing Then
					_ExtraColumns = New Dictionary(Of String, Object)()
				End If
				Return _ExtraColumns
			End Get
			Set(ByVal value As Dictionary(Of String, Object))
				_ExtraColumns = New Dictionary(Of String, Object)(value)
			End Set
		End Property
		Private _ExtraColumns As Dictionary(Of String, Object)

		Public Sub MarkAsDeleted()
			Me.esRowState = "Deleted"
		End Sub

		Private Sub SetDirty(ByVal [property] As String)
			If Not ModifiedColumns.Contains([property]) Then
				ModifiedColumns.Add([property])
			End If
			
			If Me.esRowState = "Unchanged" Then
				Me.esRowState = "Modified"
			End If
		End Sub
		
	End Class
	
	<XmlType(TypeName:="ShippersQueryProxyStub")> _	
	<DataContract(Name:="ShippersQuery", Namespace:= "http://www.entityspaces.net")> _	
	Partial Public Class ShippersQueryProxyStub 
		Inherits esDynamicQuerySerializable

		Public Sub New()
		
		End Sub

		Public Sub New(ByVal joinAlias As String)
			MyBase.es.JoinAlias = joinAlias
		End Sub	
		
		
		Protected Overrides Function GetQueryName() As String
			Return "ShippersQuery"
		End Function

		#Region "Implicit Casts"

		Public Shared Widening Operator CType(ByVal query As ShippersQueryProxyStub) As String
			Return ShippersQueryProxyStub.SerializeHelper.ToXml(query)
		End Operator

		#End Region
		

		Public ReadOnly Property ShipperID As esQueryItem
			Get
				Return new esQueryItem(Me, "ShipperID", esSystemType.Int32)
			End Get
		End Property 
		
		Public ReadOnly Property CompanyName As esQueryItem
			Get
				Return new esQueryItem(Me, "CompanyName", esSystemType.String)
			End Get
		End Property 
		
		Public ReadOnly Property Phone As esQueryItem
			Get
				Return new esQueryItem(Me, "Phone", esSystemType.String)
			End Get
		End Property 
		
	
	End Class
	
End Namespace

File: Tests/CSharp/TestSqlServer35/TestSqlServer35/BusinessObjects/ClientProxies/Northwind/Shippers.cs
/*
===============================================================================
                    EntitySpaces Studio by EntitySpaces, LLC
             Persistence Layer and Business Objects for Microsoft .NET
             EntitySpaces(TM) is a legal trademark of EntitySpaces, LLC
                          http://www.entityspaces.net
===============================================================================
EntitySpaces Version : 2012.1.0319.0
EntitySpaces Driver  : SQL
Date Generated       : 3/17/2012 4:39:40 AM
===============================================================================
*/

using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Text;
using System.Xml.Serialization;
using System.Runtime.Serialization;

using EntitySpaces.DynamicQuery;

namespace Proxies
{
	[DataContract(Namespace = "http://tempuri.org/", Name = "ShippersCollection")]	
	[XmlType(TypeName = "ShippersCollectionProxyStub")]	
	[Serializable]	
	public partial class ShippersCollectionProxyStub 
	{
		public ShippersCollectionProxyStub() { }
	
		[DataMember(Name = "Collection", EmitDefaultValue = false)]
		public BindingList<ShippersProxyStub> Collection = new BindingList<ShippersProxyStub>();
		
		public bool IsDirty()
		{
			foreach (ShippersProxyStub obj in Collection)
			{
				if (obj.IsDirty()) return true;
			}

			return false;
		}
	}
	
	[DataContract(Namespace = "http://tempuri.org/", Name = "Shippers")]
	[XmlType(TypeName = "ShippersProxyStub")]	
	[Serializable]	
	public partial class ShippersProxyStub : object, System.ComponentModel.INotifyPropertyChanged
	{
		public ShippersProxyStub()
		{
			this.esRowState = "Added";
		}
		
		public bool IsDirty()
		{
			return esRowState != "Unchanged" ? true : false;
		}
		
		public event System.ComponentModel.PropertyChangedEventHandler PropertyChanged;
		
		protected void RaisePropertyChanged(string propertyName) 
		{
			System.ComponentModel.PropertyChangedEventHandler propertyChanged = this.PropertyChanged;
			if ((propertyChanged != null)) 
			{
				propertyChanged(this, new System.ComponentModel.PropertyChangedEventArgs(propertyName));
			}
		}
		
		[DataMember(Name="a0", Order=1, EmitDefaultValue=false)]
		public System.Int32? ShipperID
		{
			get
			{
				return _ShipperID;
			}
			set
			{
				if (_ShipperID != value)
				{
					_ShipperID = value;
					SetDirty("ShipperID");
					RaisePropertyChanged("ShipperID");
				}
			}
		}
		private System.Int32? _ShipperID;
		
		[DataMember(Name="a1", Order=2, EmitDefaultValue=false)]
		public System.String CompanyName
		{
			get
			{
				return _CompanyName;
			}
			set
			{
				if (_CompanyName != value)
				{
					_CompanyName = value;
					SetDirty("CompanyName");
					RaisePropertyChanged("CompanyName");
				}
			}
		}
		private System.String _CompanyName;
		
		[DataMember(Name="a2", Order=3, EmitDefaultValue=false)]
		public System.String Phone
		{
			get
			{
				return _Phone;
			}
			set
			{
				if (_Phone != value)
				{
					_Phone = value;
					SetDirty("Phone");
					RaisePropertyChanged("Phone");
				}
			}
		}
		private System.String _Phone;
		
	
		[DataMember(Name="a10000", Order=10000)]
		public string esRowState
		{
			get { return _esRowState; }
			set 
			{
				if(_esRowState != value)
				{
					_esRowState = value; 
					RaisePropertyChanged("esRowState");
				}
			}
		}
		private string _esRowState = "Unchanged";
		
		[DataMember(Name="a10001", Order=10001, EmitDefaultValue=false)]
		public List<string> ModifiedColumns
		{
			get
			{
				if (_ModifiedColumns == null)
				{
					_ModifiedColumns = new List<string>();
				}
				return _ModifiedColumns;
			}
			set
			{
				_ModifiedColumns = new List<string>(value);
			}
		}
		List<string> _ModifiedColumns;
		
		[DataMember(Name="a10002", Order=10002, EmitDefaultValue=false)]		
		public Dictionary<string, object> esExtraColumns
		{
			get
			{
				if (_ExtraColumns == null)
				{
					_ExtraColumns = new Dictionary<string, object>();
				}
				return _ExtraColumns;
			}
			set
			{
				_ExtraColumns = new Dictionary<string, object>(value);
			}
		}
		Dictionary<string, object> _ExtraColumns;

		public void MarkAsDeleted()
		{
			this.esRowState = "Deleted";
		}

		private void SetDirty(string property)
		{
			if (!ModifiedColumns.Contains(property))
			{
				ModifiedColumns.Add(property);
			}

			if (this.esRowState == "Unchanged")
			{
				this.esRowState = "Modified";
			}
		}
	}
	
	[XmlType(TypeName = "ShippersQueryProxyStub")]	
	[Serializable]	
	[DataContract(Name = "ShippersQuery", Namespace = "http://www.entityspaces.net")]
	public partial class ShippersQueryProxyStub : esDynamicQuerySerializable
	{
		public ShippersQueryProxyStub() { }
		
		public ShippersQueryProxyStub(string joinAlias)
		{
			this.es.JoinAlias = joinAlias;
		}	
		
		override protected string GetQueryName()
		{
			return "ShippersQuery";
		}
		
		#region Implicit Casts

		public static implicit operator string(ShippersQueryProxyStub query)
		{
			return Proxies.ShippersQueryProxyStub.SerializeHelper.ToXml(query);
		}

		#endregion
	
		public esQueryItem ShipperID
		{
			get { return new esQueryItem(this, "ShipperID", esSystemType.Int32); }
		} 
		
		public esQueryItem CompanyName
		{
			get { return new esQueryItem(this, "CompanyName", esSystemType.String); }
		} 
		
		public esQueryItem Phone
		{
			get { return new esQueryItem(this, "Phone", esSystemType.String); }
		} 
		
	}
}